modules/apps/journal/journal-content-web/src/main/java/com/liferay/journal/content/web/internal/display/context/JournalContentDisplayContext.java

```diff
@@ -358,9 +360,37 @@ public String getDDMTemplateKey() throws PortalException {
 			return _ddmTemplateKey;
 		}
 
-		String ddmTemplateKey = ParamUtil.getString(
-			_portletRequest, "ddmTemplateKey",
-			_journalContentPortletInstanceConfiguration.ddmTemplateKey());
+		String ddmTemplateKey = null;
+
+		if (FeatureFlagManagerUtil.isEnabled(
+				_themeDisplay.getCompanyId(), "LPD-27566")) {
+
+			String ddmTemplateExternalReferenceCode =
+				_journalContentPortletInstanceConfiguration.
+					ddmTemplateExternalReferenceCode();
+
+			ddmTemplateKey = ParamUtil.getString(
+				_portletRequest, "ddmTemplateKey");
+
+			if (Validator.isNotNull(ddmTemplateExternalReferenceCode) &&
+				Validator.isNull(ddmTemplateKey)) {
+
+				DDMTemplate ddmTemplate =
+					_ddmTemplateLocalService.
+						fetchDDMTemplateByExternalReferenceCode(
+							ddmTemplateExternalReferenceCode,
+							_themeDisplay.getScopeGroupId());
+
+				if (ddmTemplate != null) {
+					ddmTemplateKey = ddmTemplate.getTemplateKey();
+				}
+			}
+		}
+		else {
+			ddmTemplateKey = ParamUtil.getString(
+				_portletRequest, "ddmTemplateKey",
+				_journalContentPortletInstanceConfiguration.ddmTemplateKey());
+		}
```

Comment:
I would simply further (optional):

```java
		String ddmTemplateKey =  ParamUtil.getString(
				_portletRequest, "ddmTemplateKey");

		if (Validator.isNull(ddmTemplateKey)) {

			// All this IF-ELSE can be moved to a private method like _getConfigurationDDMTemplateKey() and it can be reused in isDefaultTemplate()
			if (FeatureFlagManagerUtil.isEnabled(
					_themeDisplay.getCompanyId(), "LPD-27566")) {

				String ddmTemplateExternalReferenceCode =
					_journalContentPortletInstanceConfiguration.
						ddmTemplateExternalReferenceCode();

				if (Validator.isNotNull(ddmTemplateExternalReferenceCode)) {

					DDMTemplate ddmTemplate =
						_ddmTemplateLocalService.
							fetchDDMTemplateByExternalReferenceCode(
								ddmTemplateExternalReferenceCode,
								_themeDisplay.getScopeGroupId());

					if (ddmTemplate != null) {
						ddmTemplateKey = ddmTemplate.getTemplateKey();
					}
				}
			}
			else {
				ddmTemplateKey = _journalContentPortletInstanceConfiguration.ddmTemplateKey();
			}
		}

```

Reply:
I think this is not worth it because in isDefaultTemplate we dont need to get the template key to apply some logic , only we need to know if there is a template configured, this means if we have something in _journalContentPortletInstanceConfiguration.ddmTemplateExternalReferenceCode() , so, the code have different purpose, but thanks for the recommendation.